spec/support/test_factories.rb

```diff
@@ -0,0 +1,41 @@
+def add_topic
```

Comment:
This file is a good start for reusing some common code within feature specs.

It'd be good to have these methods namespaced so they're within a module. Currently the methods are in the global namespace and this leaves them more exposed to naming collisions than needed. 

For example, these methods are available to model and controller specs when they're needed only by feature specs.

We can reduce their exposure by introducing a module that is configured to be included in feature specs. The pattern looks like this:

```ruby
# rename file to feature_spec_helper.rb
module FeatureSpecHelper

  # Move methods into the FeatureSpecHelper module
  def add_topic
    ...
  end

  def fill_out_credentials
    ...
  end
  
  # ...and the other existing methods go in this module too
end

# This goes in the same file, feature_spec_helper.rb:
RSpec.configure do |config|
  # Configure Rspec to include the module in feature specs:
  config.include FeatureSpecHelper, type: :feature
end
```

Please get in touch with any questions.